### PR TITLE
[FLINK-36756][sql-gateway] Bump up the sql gateway rest version

### PR DESCRIPTION
**Pull request by fsk119**
## What is the purpose of the change

*Bump up the SQL Gateway REST API verison.*

### CODE REVIEW

**Comment by davidradl** [comment]
Reviewed by Chi on 28/11/24. Asked submitter questions

File: flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/rest/util/SqlGatewayRestAPIVersion.java [hunk 1]
```diff
@@ -51,7 +51,10 @@ public enum SqlGatewayRestAPIVersion
     V2(false, true),
 
     // V3 introduces materialized table related APIs
-    V3(true, true);
+    V3(false, true),
+
+    // V4 supports to deploy script to application cluster
+    V4(true, true);
```
[the remark this answers is not in the document]
Comment (fsk119) [reply flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/rest/util/SqlGatewayRestAPIVersion.java:57]: AFAIK, Hive[1] or Livy[2] support to bump its version when adding a new method. I think it brings following benefits:

* It make users more easy to use the gateway. Users knows the differences among different vesion. They don't need to read the Flink release notes to understand the differences.
* It reduces the ease that developers understand what is incompatible? Every REST API change should be used to bump the version.

[1] https://github.com/apache/hive/blob/master/service-rpc/if/TCLIService.thrift#L40
[2] https://github.com/apache/incubator-livy/blob/1f6bd7ab55db3e5b08ff133d554b32db83a23890/server/src/main/scala/org/apache/livy/server/ApiVersions.scala#L24

**Review by xuyangzhong — APPROVED** [review]
LGTM. Thanks for this update!

**Comment by fsk119** [comment]
Merging...

**Pull request merged**